doc/connectivity/networking/api/tls_credentials_shell.rst

```diff
@@ -0,0 +1,243 @@
+   "``<FORMAT>``", "Specifies the retrieval format for the provided credential. See :ref:`tls_credentials_shell_data_formats` for valid values."
+
+.. _tls_credentials_shell_list_cred:
+
+List Credentials (``list``)
+===========================
+
+List TLS credentials in the credential store.
+
+Usage
+-----
+
+To list all available credentials, use:
+
+.. code-block:: shell
+
+   cred list
+
+To list all credentials with a specified sectag, use:
+
+.. code-block:: shell
+
+   cred list <SECTAG>
+
+To list all credentials with a specified credential type, use:
+
+.. code-block:: shell
+
+   cred list any <TYPE>
+
+To list all credentials with a specified credential type and sectag, use:
+
+.. code-block:: shell
+
+   cred list <SECTAG> <TYPE>
+
+
+Arguments
+---------
+
+.. csv-table::
+   :header: "Argument", "Description"
+   :widths: 15 85
+
+   "``<SECTAG>``", "Optional. If provided, only list credentials with this sectag. Pass ``any`` or omit to allow any sectag. Otherwise, can be any non-negative integer."
+   "``<TYPE>``", "Optional. If provided, only list credentials with this credential type. Pass ``any`` or omit to allow any credential type. Otherwise, see :ref:`tls_credentials_shell_cred_types` for valid values."
+
+
+Output
+------
+
+The command outputs all matching credentials in the following (CSV-compliant) format:
+
+.. code-block:: shell
+
+   <SECTAG>,<TYPE>,<DIGEST>,<STATUS>
+
+Where:
+
+.. csv-table::
+   :header: "Symbol", "Value"
+   :widths: 15 85
+
+   "``<SECTAG>``", "The sectag of the listed credential. A non-negative integer."
+   "``<TYPE>``", "Credential type short-code (see :ref:`tls_credentials_shell_cred_types` for details) of the listed credential."
+   "``<DIGEST>``", "A string digest representing the credential contents. The exact nature of this digest may vary depending on credentials storage backend, but currently for all backends this is a base64 encoded SHA256 hash of the raw credential contents (so different storage formats for essentially identical credentials will have different digests)."
+   "``<STATUS>``", "Status code indicating success or failure with generating a digest of the listed credential. 0 if successful, negative error code specific to the storage backend otherwise. Lines for which status is not zero will be printed with error formatting."
+
+After the list is printed, a final summary of the found credentials will be printed in the form:
+
+.. code-block:: shell
+
+   <N> credentials found.
+
+Where `<N>` is the number of credentials found, and is zero if none are found.
+
+.. _tls_credentials_shell_cred_types:
+
+Credential Types
+****************
+
+The following keywords (case-insensitive) may be used to specify a credential type:
+
+.. csv-table::
+   :header: "Keyword(s)", "Meaning"
+   :widths: 15 85
+
+   "``CA_CERT``, ``CA``", "A trusted CA certificate."
+   "``SERVER_CERT``, ``SELF_CERT``, ``SELF``, ``SERV``", "Self or server certificate."
```

Comment:
What about calling this `CLIENT_CERT` instead of `SELF_CERT`?

Reply:
I can do that -- I like `SELF` personally, since this, IMO, is the most accurate description, since the TLS client can function both as a server and a client, and in both cases the cert represents "us" or "our self".

So I could add `CLIENT` as a third synonym, unless there is something I'm forgetting that means `SELF` is problematic to keep around as a synonym`

Reply:
It doesn't matter to me if you keep `SELF`, but please add `CLIENT`.